### DISCUSSION TITLE
NDM should never bind BD with empty bd tag value to any BDC

### DISCUSSION THREAD

**Discussion by sonasingh46** [openebs/openebs#3139] [General] [answered]
A BD if tagged with an empty value should never be bound and this restriction should be enforced from the NDM.
This can help ensure that any other component(e.g cStor ) that utilizes NDM for disk management and discovery is not able to use a disk if someone wants to reserve it.
Although cStor does not uses BD with an empty tag, this should be enforced from the backend NDM

**Comment by sonasingh46** [comment]
@kmova @akhilerm -- What are your thoughts on this one?

**Reply by akhilerm** [reply]
A check  can be added in NDM operator such that a blockdevice resource with `openebs.io/block-device-tag=` can never be claimed.
The check can be added [here](https://github.com/openebs/node-disk-manager/blob/287edbe59a6a9a48d897799c204822aa79afe29d/pkg/select/blockdevice/filters.go#L265) in the filtering logic.

**Comment by akhilerm** [comment, marked as answer]
https://github.com/openebs/node-disk-manager/pull/500 has been merged which contains the changes for implementing this feature. 

The new changes includes:
- blockdevices with an empty block-device-tag value can never be claimed.
